scipy/signal/tests/test_ltisys.py

```diff
@@ -8,11 +8,123 @@
 from scipy.signal.ltisys import (ss2tf, tf2ss, lsim2, impulse2, step2, lti,
                                  bode, freqresp, lsim, impulse, step,
                                  abcd_normalize, place_poles,
-                                 TransferFunction, StateSpace, ZerosPolesGain)
+                                 TransferFunction, StateSpace, ZerosPolesGain,
+                                 controllability_matrix,
+                                 observability_matrix, controller_hessenberg)
+
 from scipy.signal.filter_design import BadCoefficients
 import scipy.linalg as linalg
 
 
+class TestCtrbObsv:
+    # The output from controller_hessenberg has been tested with random
+    # controllable and uncontrollable system against Octave which uses tb01ud
+    # from Slicot and all results were similar using the default tolerances of
+    # `allclose`.
+
+    def test_hessenberg(self):
+        # use np.allclose tolerances, atol = 0 from assert_allclose is
+        # unrealistic here especially for the comparison between
+        # np.linalg.inv(h_f.P) and h_f.P.T
+
+        # Test from the book the algorithm comes from (see [1] in docstring)
+        # it gives us references values to test against what we compute (p 177)
+        A = np.array([[1, 1, 1], [1, 1, 1], [0, 0, 1]])
+        B = np.array([[1, 1], [1, 1], [1, 1]])
+
+        ref_Abar = np.array([[2.3333, -0.4714, 0], [0.9428, 0.6667, 0],
+                             [0, 0, 0]])
```

Comment:
... and then this can just be `ref_Abar = [[...]]`, i.e. no `np.array()` around it.